### ISSUE TITLE
Error uploading files - 503 Service Unavailable

### ISSUE THREAD

**Issue by leetosc** [directus/directus#6946]
### Describe the Bug

I am running in docker using the `docker-compose` file on this page, unchanged https://docs.directus.io/guides/installation/docker/
Running behind a nginx reverse proxy.

<details>
<summary>nginx conf</summary>

```nginx
server {
    server_name directus.mydomain.com;


    location / {
      proxy_pass http://localhost:8055/;
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection 'upgrade';
      proxy_set_header Host $host;
      proxy_cache_bypass $http_upgrade;
    }

    client_max_body_size 900M;


    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/directus.mydomain.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/directus.mydomain.com/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}

server {
    if ($host = directus.mydomain.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


    server_name directus.mydomain.com;
    listen 80;
    return 404; # managed by Certbot


}
```

</details>

When uploading a file, it gives this error: 
![image](https://user-images.githubusercontent.com/13279967/126858095-f6dbb08b-e599-4527-9bd2-c1ab3eed76e7.png)

```
{
  "errors": [
    {
      "message": "Couldn't save file 9c418ffc-b985-47cf-a84b-d23c1e6f8e5d..png",
      "extensions": {
        "code": "SERVICE_UNAVAILABLE",
        "service": "files"
      }
    }
  ]
}
```
docker logs show this:
```
05:03:38 ⚠️  Couldn't save file 28f53a19-092f-49b7-893c-9268ce0bc96a..png
05:03:38 ⚠️  E_UNKNOWN: An unknown error happened with the file 28f53a19-092f-49b7-893c-9268ce0bc96a..png.

Error code: EACCES
Original stack:
Error: EACCES: permission denied, open '/directus/uploads/28f53a19-092f-49b7-893c-9268ce0bc96a..png'
05:03:38 ✨ request errored POST 503 /files 15ms
```

### To Reproduce

File Library -> Upload a file -> POST request returns 503.

### What version of Directus are you using?

9.0.0-rc.83

### What version of Node.js are you using?

v16.4.1

### What database are you using?

postgres:12

### What browser are you using?

Chrome

### What operating system are you using?

ubuntu

### How are you deploying Directus?

Docker

**Comment by paescuj** [comment]
Fixed in v9.0.0-rc84 via #6584.
Can you please upgrade and recheck?

**Comment by r1cebank** [comment]
I am encountering the same issue with the Docker build, after deploying rc85, the extra dot in the filename is fixed but file upload still fails:

```
Error code: EACCES
Original stack:
Error: EACCES: permission denied, open '/directus/uploads/dc648d2d-9318-4451-80c6-c177d6a120cc.JPG'
```

**Comment by leetosc** [comment]
> Fixed in v9.0.0-rc84 via #6584.
> Can you please upgrade and recheck?

I edited the `docker-compose.yml` to use this image `image: directus/directus:9.0.0-rc.85` instead of `image: directus/directus:latest` (which is rc83)

Now the entire app is showing a blank white screen with these errors in the console:
![image](https://user-images.githubusercontent.com/13279967/126890430-c4238091-11f7-4cb1-9287-a898a960f7a6.png)

the docker image for rc84 also gives the same result

**Comment by paescuj** [comment]
> > Fixed in v9.0.0-rc84 via #6584.
> > Can you please upgrade and recheck?
> 
> I edited the `docker-compose.yml` to use this image `image: directus/directus:9.0.0-rc.85` instead of `image: directus/directus:latest` (which is rc83)
> 
> Now the entire app is showing a blank white screen with these errors in the console:
> ![image](https://user-images.githubusercontent.com/13279967/126890430-c4238091-11f7-4cb1-9287-a898a960f7a6.png)
> 
> the docker image for rc84 also gives the same result

You need to set the environment variable `PUBLIC_URL` to match your public domain. E.g. in `docker-compose.yml`:
```yaml
    environment:
      PUBLIC_URL: 'https://directus.mydomain.com'
```

**Comment by paescuj** [comment]
> I am encountering the same issue with the Docker build, after deploying rc85, the extra dot in the filename is fixed but file upload still fails:
> 
> ```
> Error code: EACCES
> Original stack:
> Error: EACCES: permission denied, open '/directus/uploads/dc648d2d-9318-4451-80c6-c177d6a120cc.JPG'
> ```
> 
> after some digging, the upload directory appears to be owned by root
> 
> ```
> drwxr-xr-x    2 node     node          4096 Jul 15 19:41 database
> drwxr-xr-x    8 node     node          4096 Jul 25 04:36 extensions
> drwxr-xr-x  595 node     node         20480 Jul 15 19:41 node_modules
> -rw-r--r--    1 node     node          1215 Jul 15 19:39 package.json
> drwxr-xr-x    2 root     root          4096 Jun 12 17:00 uploads
> ```
> 
> Could be related to this: [docker/for-mac#5480](https://github.com/docker/for-mac/issues/5480) *Note even this is reported for docker for mac project, but there are people experiencing this on WSL and ubuntu as well.

In [v9.0.0-rc.71](https://github.com/directus/directus/blob/main/changelog.md#v900-rc71-june-2-2021) the Docker image has been updated to run as user `node` instead of `root`. Therefore you need to adjust the owner of the `uploads` directory.
See https://github.com/directus/directus/discussions/6480#discussioncomment-917708 for instructions.

**Comment by rijkvanzanten** [comment]
I just setup a fresh project using the docker-compose template from the docs. I substituted `directus/directus:latest` with `directus/directus:9.0.0-rc.86` to make sure it wasn't using an older cached image. It seems to work as expected. Please make sure you're on the latest version, and be aware of https://github.com/directus/directus/discussions/6480#discussioncomment-917708 if you upgraded from an older (existing) version

**Comment by rijkvanzanten** [comment]
Thank you for taking the time to submit this! Unfortunately, our team was not been able to reproduce it based on the information provided. Therefore, we will close this ticket for now, but will happily re-open it if new details are provided that allow us to reliably replicate the problem.

As a reminder:
* Provide as much detail as possible in the issue description
* Video screen captures and clear steps are very helpful
* A full SQL dump is the fastest way for us to test/resolve (can be sent privately)
* Make sure you are on the latest version of Directus
* Did you try turning it off and on again?

---

_This is an automated response._

**Comment by leetosc** [comment]
> > > Fixed in v9.0.0-rc84 via #6584.
> > > Can you please upgrade and recheck?
> > 
> > 
> > I edited the `docker-compose.yml` to use this image `image: directus/directus:9.0.0-rc.85` instead of `image: directus/directus:latest` (which is rc83)
> > Now the entire app is showing a blank white screen with these errors in the console:
> > the docker image for rc84 also gives the same result
> 
> You need to set the environment variable `PUBLIC_URL` to match your public domain. E.g. in `docker-compose.yml`:
> 
> ```yaml
>     environment:
>       PUBLIC_URL: 'https://directus.mydomain.com'
> ```

Adding the `PUBLIC_URL` environment variable fixed this for me. 
I ran into the permissions issue and running `docker-compose exec -u root <directus-service-name>chown -R node:node /directus/database /directus/extensions /directus/uploads` fixed that.

It is working now, thanks!

**Comment by paescuj** [comment]
@leetosc Thanks for your feedback!

**Comment by arladmin** [comment]
Hi all.
I'm facing the same issue.

Error:
```
15:34:48 ⚠️  Couldn't save file 8f7e5c71-d0f8-48ef-ad71-6f708f8a3c41.jpg
15:34:48 ⚠️  E_UNKNOWN: An unknown error happened with the file 8f7e5c71-d0f8-48ef-ad71-6f708f8a3c41.jpg.

Error code: EACCES
Original stack:
Error: EACCES: permission denied, open '/directus/uploads/8f7e5c71-d0f8-48ef-ad71-6f708f8a3c41.jpg'
15:34:48 ✨ request errored POST 503 /files 29ms
```

My setup:
Docker based, directus v9.0.0-rc.87
Database: Postgres 12

docker-compose:
```
version: '3.2'
services:
  database:
    container_name: database
    image: postgres:12
    volumes:
      - ./data/database:/var/lib/postgresql/data
    networks:
      - directus
    environment:
      POSTGRES_USER: 'directus'
      POSTGRES_PASSWORD: 'directus'
      POSTGRES_DB: 'directus'
    ports:
      - 5432:5432

  cache:
    container_name: cache
    image: redis:6
    networks:
      - directus

  directus:
    container_name: directus
    image: directus/directus
    ports:
      - 8055:8055
    networks:
      - directus
    depends_on:
      - cache
      - database
    env_file:
      - .env
    volumes:
      - ./data/extensions:/directus/extensions
      - ./data/uploads:/directus/uploads
      - ./data/database:/directus/database
```

```.env``` parameters:
````
KEY=key
SECRET=secret

DB_CLIENT=pg
DB_HOST=database
DB_PORT=5432
DB_DATABASE=directus
DB_USER=directus
DB_PASSWORD=directus

CACHE_ENABLED=true
CACHE_STORE=redis
CACHE_REDIS=redis://cache:6379
CACHE_AUTO_PURGE="true"

ADMIN_EMAIL=email
ADMIN_PASSWORD=password

PUBLIC_URL=https://mydomain
```

**Comment by meghraitudu** [comment]
> > > > Fixed in v9.0.0-rc84 via #6584.
> > > > Can you please upgrade and recheck?
> > > 
> > > 
> > > I edited the `docker-compose.yml` to use this image `image: directus/directus:9.0.0-rc.85` instead of `image: directus/directus:latest` (which is rc83)
> > > Now the entire app is showing a blank white screen with these errors in the console:
> > > the docker image for rc84 also gives the same result
> > 
> > 
> > You need to set the environment variable `PUBLIC_URL` to match your public domain. E.g. in `docker-compose.yml`:
> > ```yaml
> >     environment:
> >       PUBLIC_URL: 'https://directus.mydomain.com'
> > ```
> 
> Adding the `PUBLIC_URL` environment variable fixed this for me. I ran into the permissions issue and running `docker-compose exec -u root <directus-service-name>chown -R node:node /directus/database /directus/extensions /directus/uploads` fixed that.
> 
> It is working now, thanks!

this worked for me too. :)

**Comment by g05357005** [comment]
chown to node:node works for me too.


10.6.2 still have this problem.

**Comment by hermann-san** [comment]
@leetosc  [suggestion](https://github.com/directus/directus/issues/6946#issuecomment-887998767) has also fixed it for me. Thanks for that

Instead of docker compose I've used docker compose (without dash)
Prerequisite is that volumes are configured in docker-compose.yml with ./uploads + ./extentions (not uncommented with #)
`cd [directus home]`
The service name is in docker-compose.yml  and there are 3 services I've found there.  "database", "cache" and "directus"
"directus" was the service name for me.

It would be nice to add this info to the doc's



docker-compose exec -u root <directus-service-name>chown -R node:node /directus/database /directus/extensions /directus/uploads
